src/test/java/com/list/todo/it/TagTest.java

```diff
@@ -0,0 +1,510 @@
+        //arrange
+        TagInput tagInput = new TagInput("Home", "ff");
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        when(tagServiceMock.addTag(tagInput, CURRENT_USER_ID)).thenReturn(Optional.of(new Tag("Home", CURRENT_USER_ID, "ff")));
+
+        //act, assert
+        this.mockMvc.perform(post("/api/tags")
+                .content(objectMapper.writeValueAsString(tagInput))
+                .contentType(MediaType.APPLICATION_JSON_UTF8))
+                .andDo(print())
+                .andExpect(jsonPath("tagName").value(tagInput.getTagName()))
+                .andExpect(jsonPath("ownerId").value(CURRENT_USER_ID))
+                .andExpect(status().isOk());
+
+        verify(tagServiceMock, times(1)).addTag(tagInput, CURRENT_USER_ID);
+    }
+
+    @Test
+    public void updateTag_OnExistentTag_ReturnsAObjectOfTag() throws Exception {
+        //arrange
+        Tag tag = new Tag("Home", CURRENT_USER_ID, "ff");
+        tag.setId(2L);
+
+        Tag updatedTag = new Tag("Job", CURRENT_USER_ID, "ff");
+        updatedTag.setId(2L);
+
+        TagInput tagInput = new TagInput("Job", "ff");
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        when(tagServiceMock.getTagById(tag.getId())).thenReturn(Optional.of(tag));
+        when(tagServiceMock.updateTag(tag.getId(), tagInput)).thenReturn(Optional.of(updatedTag));
+
+        //act, assert
+        this.mockMvc.perform(put("/api/tags/{id}", "2")
+                .content(objectMapper.writeValueAsString(tagInput))
+                .contentType(MediaType.APPLICATION_JSON_UTF8))
+                .andDo(print())
+                .andExpect(jsonPath("id").value(updatedTag.getId()))
+                .andExpect(jsonPath("tagName").value(tagInput.getTagName()))
+                .andExpect(jsonPath("ownerId").value(CURRENT_USER_ID))
+                .andExpect(status().isOk());
+
+        verify(tagServiceMock, times(1)).getTagById(tag.getId());
+        verify(tagServiceMock, times(1)).updateTag(tag.getId(), tagInput);
+    }
+
+    @Test
+    public void updateTag_OnNonExistentTag_ReturnsANotFound() throws Exception {
+        //arrange
+        Long tagId = 1000L;
+        when(tagServiceMock.getTagById(tagId)).thenReturn(Optional.empty());
+
+        TagInput tagInput = new TagInput("Job", "ff");
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        //act, assert
+        this.mockMvc.perform(put("/api/tags/{id}", tagId)
+                .content(objectMapper.writeValueAsString(tagInput))
+                .contentType(MediaType.APPLICATION_JSON_UTF8))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+
+        verify(tagServiceMock, times(1)).getTagById(tagId);
+        verify(tagServiceMock, times(0)).updateTag(tagId, tagInput);
+    }
+
+    @Test
+    public void updateTag_OnAnotherUser_ReturnsAIsForbidden() throws Exception {
+        //arrange
+        Long ownerId = 10L;
```

Comment:
You can move ownerId, tagId, taksId etc. to constants to avoid creating local variables every time